lib/src/context.rs

```diff
@@ -254,7 +254,11 @@ fn send_mode_to_config_head(
         let min = refresh - 501;
         let max = refresh + 501;
 
-        if let Some(mode) = mode_iter().find(|mode| min < mode.refresh && max > mode.refresh) {
+        let mode = mode_iter()
+            .find(|mode| mode.refresh == refresh)
+            .or_else(|| mode_iter().find(|mode| min < mode.refresh && max > mode.refresh));
```

Comment:
maybe we could also do something like `min_by_key(|mode| (mode.refresh - refresh).abs())` to find the closest match?

So, if we still want to filter close values (untested):
```suggestion
            .or_else(|| mode_iter()
                .filter(|mode| min < mode.refresh && max > mode.refresh)
                .min_by_key(|mode| (mode.refresh - refresh).abs())
            );
```